pandas/core/internals/blocks.py

```diff
@@ -1168,16 +1170,36 @@ def _interpolate_with_fill(
         # We only get here for non-ExtensionBlock
         fill_value = convert_scalar(self.values, fill_value)
 
-        values = missing.interpolate_2d(
-            values,
-            method=method,
-            axis=axis,
-            limit=limit,
-            fill_value=fill_value,
-            dtype=self.dtype,
-        )
+        # We have to distinguish two cases:
+        # 1. When kwarg `limit_area` is used: It is not
+        #    supported by `missing.interpolate_2d()`. Using this kwarg only
+        #    works by applying the fill along a certain axis.
+        # 2. All other cases: Then, `missing.interpolate_2d()` can be used.
+        if limit_area is not None:
+
+            def func(x):
```

Comment:
instead of handling this case here and distributing the logic, can you simply call it from interpolate_2d?

Reply:
Yes. Will be moved in next series of commits.